virtual_rainforest/models/animals/animal_traits.py

```diff
@@ -0,0 +1,82 @@
+"""The `models.animals.animal_traits` module contains classes that organize
+animal traits into enumerations for use by the Functional Group class in the
+:mod:`~virtual_rainforest.models.animals.functional_group` module.
+"""  # noqa: D205, D415
+
+from enum import Enum, auto
+
+
+class MetabolicType(Enum):
+    """Enumeration for metabolic types."""
+
+    ENDOTHERMIC = auto()
+    ECTOTHERMIC = auto()
+
+    @classmethod
+    def from_str(cls, value: str) -> "MetabolicType":
+        """Create a `MetabolicType` enumeration member from its string representation.
+
+        Args:
+            value: The string representation of the metabolic type.
+
+        Returns:
+            The corresponding `MetabolicType` enumeration member.
+
+        Raises:
+            ValueError: If the provided string does not match any valid metabolic type.
+        """
+        try:
+            return cls[value.upper()]
+        except KeyError:
+            raise ValueError(f"Invalid metabolic type: {value}")
```

Comment:
Mmmmm... I sorry for this. I've just noticed that the `StrEnum` I used in my example is only available from Python 3.11. Being that the case, you can explicitly set the values instead of using `auto`:

```suggestion
class MetabolicType(Enum):
    """Enumeration for metabolic types."""

    ENDOTHERMIC = "endothermic"
    ECTOTHERMIC = "ectothermic"
```

This should work fine when calling `MetabolicType("endothermic")`.